src/web/app/components/question-submission-form/question-submission-form.component.ts

```diff
@@ -258,4 +333,53 @@ export class QuestionSubmissionFormComponent implements DoCheck {
     this.responsesSave.emit(this.model);
   }
 
+  /**
+   * Returns the respective {@code FeedbackRecipientLabelType}
+   */
+  getSelectionLabelType(recipientType: FeedbackParticipantType): FeedbackRecipientLabelType {
+    switch (recipientType) {
+      case FeedbackParticipantType.STUDENTS:
+      case FeedbackParticipantType.STUDENTS_EXCLUDING_SELF:
+        return FeedbackRecipientLabelType.INCLUDE_SECTION;
+      case FeedbackParticipantType.STUDENTS_IN_SAME_SECTION:
+        return FeedbackRecipientLabelType.INCLUDE_TEAM;
+      default:
+        return FeedbackRecipientLabelType.INCLUDE_NAME;
+    }
+  }
+
+  /**
+   * Returns the Selection Option label as per the recipientType.
+   */
+  getSelectionOptionLabel(recipient: FeedbackResponseRecipient): string {
+    if (!this.isSectionTeamShown) {
+      return recipient.recipientName;
+    }
+
+    let label: string = recipient.recipientName;
+
+    if (recipient.recipientTeam) {
+      label = `${recipient.recipientTeam} | `.concat(label);
```

Comment:
Can make it slightly more readable:
```suggestion
      label = `${recipient.recipientTeam} | ${label}`;
```